### ISSUE TITLE
uninstall package dont load package lang file

### ISSUE THREAD

**Issue by sfs-it** [joomla/joomla-cms#14087]
### Steps to reproduce the issue
Remove a installed a package that support lang file, and pkg_script.php
on class pkg_XXX uninstall method will be called a JText::_('MESSAGE)
MESSAGE will not traslated

### Expected result
Display, the correct MESSAGE translation (get from pkg_XXX.sys.ini)

### Actual result
MESSAGE will not traslated

### System information (as much as possible)
Joomla 3.6.5

### Additional comments
i wrote this patch and committed on forum.joomla.org
https://forum.joomla.org/viewtopic.php?f=715&t=946981&p=3460994#p3460994

on line 722 of libraries/cms/installer/installer.php
```
                // Add the languages if installed with package
                if ($type === 'package')
                {
                    if ($this->extension->load($identifier,null))
                    {
                        $lang = JFactory::getLanguage();
                        $lang_tag = $lang->getTag();

                        if (!$lang->load($this->extension->element.'.sys',JPATH_SITE,$lang_tag))
                        {
                            $lang->load($this->extension->element.'.sys',JPATH_SITE);
                        }
                    }
                }

```

--
Agostino Zanutto
SFS.it

**Comment by Bakual** [comment]
Since you already have code proposed, can you please make a PR so it can be reviewed and tested?
See https://docs.joomla.org/Using_the_Github_UI_to_Make_Pull_Requests for some guidance how to create a PR

**Comment by sfs-it** [comment]
@Bakual Sorry it's my first time :D
Reading code of 3.7 on line 722 of the same file there is a comment that explain that "adapter" will translate language... and there isnt loaded.
https://github.com/joomla/joomla-cms/blob/3.7.x/libraries/cms/installer/installer.php#L722
Later I try to create a devel host 3.7 based and I'll see if the issue die on 3.6.5.

Have a good day.

**Comment by infograf768** [comment]
I would need an extension which has such a script to test.

**Comment by franz-wohlkoenig** [comment]
@sfs-it do you have such an Extension?

**Comment by sfs-it** [comment]
@franz-wohlkoenig you fixed it on 3.6.5 or 3.7?
I can build an extension dedicated if you need for it.

**Comment by franz-wohlkoenig** [comment]
@sfs-it @infograf768 was asking for an Extension, guessing 3.7

**Comment by sfs-it** [comment]
Gimme 20' and I build one for testing.

**Comment by sfs-it** [comment]
I placed 2 empty install packages.
fist one is for a system plugin => https://www.sfs.it/downloads/J36-plg_system_testing_uninstall_demo.zip
second is a template =>  https://www.sfs.it/downloads/J36-tpl_testing_uninstall_demo.zip

both had all events enabled on script.php.

Template uninstall dont generate any script.php call on 3.6.5.

Attachments:
[J36-plg_system_testing_uninstall_demo.zip](https://github.com/joomla/joomla-cms/files/896425/J36-plg_system_testing_uninstall_demo.zip)
[J36-tpl_testing_uninstall_demo.zip](https://github.com/joomla/joomla-cms/files/896426/J36-tpl_testing_uninstall_demo.zip)

**Comment by franz-wohlkoenig** [comment]
@infograf768 can you please have a look above?

**Comment by infograf768** [comment]
I fail to understand the issue as well as the code proposed.

The proposed plugin (its type is NOT `package` but `plugin`) installs/uninstalls OK and the uninstall message displays OK, i.e., as I changed the string
`PLG_SYSTEM_TESTING_UNINSTALL_DEMO_UNINSTALL_TEXT="Uninstallation went perfectly"`
this without the patch.

It does not work for the template with or without patch, no message at all, but I guess this may be due to your code rather than core.

**Comment by mbabker** [comment]
The template install adapter is only on the standardized install path.  So install actions should trigger the scriptfile.

The uninstall path will be fixed in 4.0.  Mainly because standardizing all of that for all of the adapters will most require B/C breaks in the system internals.

**Comment by sfs-it** [comment]
Sorry... I create pkg, gimme 10'

**Comment by sfs-it** [comment]
Demo package
[J36-pkg_testing_uninstall_demo.zip](https://github.com/joomla/joomla-cms/files/900038/J36-pkg_testing_uninstall_demo.zip)

**Comment by sfs-it** [comment]
Here there are the pkg
[J36-pkg_testing_uninstall_demo.zip](https://github.com/joomla/joomla-cms/files/900052/J36-pkg_testing_uninstall_demo.zip)

**Comment by infograf768** [comment]
@sfs-it 

@mbabker has replied to you above:
https://github.com/joomla/joomla-cms/issues/14087#issuecomment-291902787

**Comment by sfs-it** [comment]
@mbabker what do you mean with "The template install adapter is only on the standardized install path. So install actions should trigger the scriptfile."
I've write an template and if I uninstall it, the class in script file arent call...

**Comment by mbabker** [comment]
Right.  The install path was standardized in 3.4.  The uninstall path hasn't yet, and as such each extension adapter still has different behaviors.

**Comment by mbabker** [comment]
https://github.com/joomla/joomla-cms/pull/16343 is the starting point for standardizing things.  It will only be 4.0 due to the required B/C breaks.

**Comment by brianteeman** [comment]
Closed so we just have one place see #16343
